src/NHibernate/Type/ManyToOneType.cs

```diff
@@ -162,11 +185,16 @@ private bool IsIdentifier(object value, ISessionImplementor session)
 
 		public override bool IsNull(object owner, ISessionImplementor session)
 		{
-			if (IsNullable && !string.IsNullOrEmpty(PropertyName))
+			if (IsNullable && !string.IsNullOrEmpty(PropertyName) && owner != null)
 			{
-				EntityEntry entry = session.PersistenceContext.GetEntry(owner);
+				var ownerPersister = session.Factory.GetEntityPersister(_entityName);
+				var id = session.GetContextEntityIdentifier(owner);
+				if (id == null)
+					return false;
+
+				var entityKey = session.GenerateEntityKey(id, ownerPersister);
 
-				return session.PersistenceContext.IsPropertyNull(entry.EntityKey, PropertyName);
+				return session.PersistenceContext.IsPropertyNull(entityKey, PropertyName);
```

Comment:
This re-uses the logic from one-to-one, as I was suggesting [here](https://github.com/nhibernate/nhibernate-core/pull/1948/files#r549147078).

Reply:
If I'm not mistaken `_entityName` is not really needed here. The following logic can be used for obtaining entity key:

```C#
private static EntityKey GetEntityKey(object owner, ISessionImplementor session)
{
	EntityEntry entry = session.PersistenceContext.GetEntry(owner);
	if (entry != null)
		return entry.EntityKey;
	if (owner is INHibernateProxy proxy)
		return session.GenerateEntityKey(proxy.HibernateLazyInitializer.Identifier, session.Factory.GetEntityPersister(proxy.HibernateLazyInitializer.EntityName));
	return null;
}
```

Reply:
This code works only if the owner belong to the same session. It will not support a detached owner or an owner belonging to another session. For merge cases, we need to support detached owners.

Reply:
`session.GetContextEntityIdentifier` doesn't support detached entity either. Same behavior - either it's present in session or it's proxy.

Reply:
> It will not support a detached owner or an owner belonging to another session.

So current fix doesn't support detached entity either. To support detached entity `ForeignKeys.GetIdentifier` needs to be used instead of `session.GetContextEntityIdentifier`.

Reply:
Anyway `IsPropertyNull` also works only for attached owners, so my point was moot, excepted for using almost the same code than `OneToOneType`.